ecs-fargate-service/ecs-fargate.tf

```diff
@@ -0,0 +1,321 @@
+  # therefor we have to create a new one before destroying the old. This also means
+  # we have to let it have a random name, and then tag it with the desired name.
+  //  lifecycle {
+  //    create_before_destroy = true
+  //  }
+  tags = merge(
+    var.tags,
+    {
+      Purpose = "Target group used for ECS service. The blue TG is used always either in ECS standard deployment or canary deployment",
+      Name    = "${var.name_prefix}-target-${var.task_container_port}"
+    }
+  )
+}
+
+# green TG is used only in canary deployment
+resource "aws_lb_target_group" "green" {
+  for_each = var.deployment_controller_type == "CODE_DEPLOY" ? toset(["canary"]) : toset([])
+  # TG name cannot be longer than 32 characters
+  name                 = format("%s%s", length(var.name_prefix) > 26 ? substr(var.name_prefix, 0, 26) : var.name_prefix, "-green")
+  vpc_id               = var.vpc_id_tg != "" ? var.vpc_id_tg : var.vpc_id
+  protocol             = var.task_container_protocol
+  port                 = var.task_container_port
+  target_type          = "ip"
+  deregistration_delay = var.deregistration_delay
+  dynamic "health_check" {
+    for_each = [var.health_check]
+    content {
+      enabled             = lookup(health_check.value, "enabled", true)
+      healthy_threshold   = lookup(health_check.value, "healthy_threshold", 2)
+      interval            = lookup(health_check.value, "interval", 30)
+      matcher             = lookup(health_check.value, "matcher", 200)
+      path                = lookup(health_check.value, "path", "/system/monitoring/common/ping")
+      port                = lookup(health_check.value, "port", "traffic-port")
+      protocol            = lookup(health_check.value, "protocol", "HTTP")
+      timeout             = lookup(health_check.value, "timeout", 15)
+      unhealthy_threshold = lookup(health_check.value, "unhealthy_threshold", 5)
+    }
+  }
+
+  # NOTE: TF is unable to destroy a target group while a listener is attached,
+  # therefor we have to create a new one before destroying the old. This also means
+  # we have to let it have a random name, and then tag it with the desired name.
+  //  lifecycle {
+  //    create_before_destroy = true
+  //  }
+
+  tags = merge(
+    var.tags,
+    {
+      Purpose = "Target group used for ECS service. The green TG is used only in canary deployment",
+      Name    = "${var.name_prefix}-target-${var.task_container_port}"
+    }
+  )
+}
+
+
+# ------------------------------------------------------------------------------
+# ECS task definition
+# ------------------------------------------------------------------------------
+
+data "aws_region" "current" {}
+
+locals {
+  task_environment = [
+    for k, v in var.task_container_environment : {
+      name  = k
+      value = v
+    }
+  ]
+
+  task_environment_secret = [
+    for k, v in var.task_container_secrets : {
+      name      = k
+      valueFrom = v
+    }
+  ]
+
```

Comment:
Would not be it better to have it as list of map since beginning in variable than do this transformation?
```
[
{k,v}, {k,v}
]
```

Perhaps it would decrease readability on module use

Reply:
yeah, IMO would be less readable